warehouse/query-core/src/main/java/datawave/query/index/lookup/BaseIndexStream.java

```diff
@@ -0,0 +1,122 @@
+package datawave.query.index.lookup;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Iterators;
+import datawave.query.jexl.visitors.JexlStringBuildingVisitor;
+import datawave.query.tables.RangeStreamScanner;
+import datawave.query.util.Tuple2;
+import org.apache.commons.jexl2.parser.JexlNode;
+
+import java.util.Iterator;
+
+/**
+ * Provides a core set of variables for the ScannerStream, Union, and Intersection.
+ *
+ * A reference to the underlying {@link datawave.query.tables.RangeStreamScanner} is required for seeking.
+ *
+ * Note that the BaseIndexStream does not implement the {@link IndexStream#seek(String)} method. Inheriting classes are responsible for determining the correct
+ * implementation.
+ */
+public abstract class BaseIndexStream implements IndexStream {
+    
+    protected RangeStreamScanner rangeStreamScanner;
+    
+    protected EntryParser entryParser;
+    
+    protected JexlNode node;
+    
+    protected StreamContext context;
+    
+    protected IndexStream debugDelegate;
+    
+    // variables to support the PeekingIterator interface
+    protected Iterator<Tuple2<String,IndexInfo>> backingIter;
+    protected Tuple2<String,IndexInfo> peekedElement;
+    protected boolean hasPeeked = false;
+    
+    public BaseIndexStream(RangeStreamScanner rangeStreamScanner, EntryParser entryParser, JexlNode node, StreamContext context, IndexStream debugDelegate) {
+        this.rangeStreamScanner = Preconditions.checkNotNull(rangeStreamScanner);
+        this.entryParser = Preconditions.checkNotNull(entryParser);
+        this.node = node;
+        this.backingIter = Iterators.transform(this.rangeStreamScanner, this.entryParser);
+        this.context = context;
+        this.debugDelegate = debugDelegate;
```

Comment:
debugDelegate seems kind of hokey.  What exactly is this supposed to do?

Reply:
Your guess is as good as mine. It's existence here is an artifact of lifting existing constructors.

Reply:
The debugDelegate is used when logging in debug mode to provide some context.